packages/middleware/src/apiClientFactory/applyContextToApi.ts

```diff
@@ -66,30 +68,75 @@ const applyContextToApi = <
   Object.entries(api).reduce(
     (prev, [callName, fn]) => ({
       ...prev,
-      [callName]: async (...args: Parameters<typeof fn>) => {
-        const transformedArgs = await hooks.before({
-          callName,
-          args,
-        });
-        const apiClientContext = {
-          ...context,
-          extendQuery: createExtendQuery(context),
+      [callName]: (() => {
+        const logger = injectHandlerMetadata(context, fn, callName);
+        /**
+         * Endpoint's handler decorated with:
+         * - hooks from every extension (hooks.before contains merged beforeCall's,
+         *   hooks.after contains merged afterCall's),
+         * - logger with metadata about scope of call (covers orchestrated parallely called endpoints),
+         * - support for building error boundary (covers orchestrated parallely called endpoints),
+         */
+        const handler = async (...args: Parameters<typeof fn>) => {
```

Comment:
What is a handler? Handler of what?

Reply:
You have multiline comment above it describing it